### PR TITLE
Stalemate

### PR DESCRIPTION
**Pull request by chris-campbell**
This method relies on the valid_move? method I forced a passed to see it function as i except and it seems to work be once valid move is merged i expect it should work.

### CODE REVIEW

File: app/models/game.rb [hunk 1]
```diff
@@ -5,16 +5,55 @@ class Game < ApplicationRecord
...
+  
+  
+  def stalemate?(color)
+    current_pieces = friendly_pieces(color)
+    possible_moves = []
+    current_pieces.each do |piece|
+      8.times do |x|
+        8.times do |y|
+          possible_moves << [x, y] if piece.valid_move?(x, y)
+        end
```
Comment (Zjohn64) [inline app/models/game.rb:17]: looks great but i think it should be piece.valid_move && !put_in_check

**Review by Zjohn64 — COMMENTED** [review]
One slight change I would make but otherwise great job Chris

File: app/models/game.rb (hunk 1 above)
Comment (chris-campbell) [reply app/models/game.rb:17]: Yeah I was trying that initially but put_in_check didn't seem to be working so I left it off until issues are resolved

Comment (chris-campbell) [reply app/models/game.rb:17]: I think the implementation of in_check? In put_in_check is off. In_check? Should take a argument

File: app/models/piece.rb [hunk 2]
```diff
@@ -177,8 +177,8 @@ def is_my_turn?
   
   def valid_move?(x,y)
     #i'm not sure I'm implementing in check correctly here. second set of eyes would be appreciated!
-    obstructed?(x,y) && off_the_board?(x,y) && in_check?(king) && not_moved_to_different_space?(x,y) && is_my_turn?
-    
+    #obstructed?(x,y) && off_the_board?(x,y) && in_check?(king) && not_moved_to_different_space?(x,y) && is_my_turn?
+    #
   end
```
Comment (ulalu) [inline app/models/piece.rb:182]: This is going to be overwritten by the valid move pr I have up right now. Should be working after that.

File: app/models/game.rb (hunk 1 above)
Comment (ulalu) [reply app/models/game.rb:17]: yep, it should take a king as the argument. I know check is broken at the moment. Zach mentioned you might have a fix @chris-campbell? If so, throw up a pr with that logic and I'd be happy to test it. If not I'll snap to it to get the check logic working. I'm sorry I didn't at the time- tests weren't working so it was a hail mary method.

File: app/models/rook.rb [hunk 3]
```diff
@@ -1,13 +1,8 @@
...
-		return false unless super.valid_move?
-		if 
-			((position_x == x) || (position_y == y)) && generic_validity(x,y)
-			return true
-		else
-	  		return false
-		end
+	  super
+			((position_x == x) || (position_y == y))
 	end
```
Comment (ulalu) [inline app/models/rook.rb:6]: great refactor

**Review by ulalu — APPROVED** [review]
Great work Chris, there's a comment in there about check logic that I'd like to connect on at some point, but I'm going to go ahead and merge this in. We'll do a quick fix to the stalemate logic once check is fixed.

File: app/models/game.rb (hunk 1 above)
Comment (chris-campbell) [reply app/models/game.rb:17]: Check has minor issues I believe. I had sent a possible solution a couple of days ago on slack but i think it was sent privately. Im not sure but my slack has been acting up had to reinstall. but if we can go over this tonight on our meet up if you'd like.

Comment (chris-campbell) [reply app/models/game.rb:17]: put_in_check? also uses a king object that isn't declared or passed in as an argument in the function.
example: current_state = king.where(color: king.color).in_check?

I expect a undefined method error for "king.where" and "king.color"

**Something like:**
_king = pieces.find_by(type: 'King', color: 'white')_ 
should return a king object to work with.

Also I think rather then king.where(), it should be something like pieces.where().

**Pull request merged**